src/ess/reduce/live/raw.py

```diff
@@ -317,19 +289,64 @@ def make_roi_filter(self) -> roi.ROIFilter:
                 indices = self._projection(indices)
         return roi.ROIFilter(indices=indices, norm=norm)
 
+    def transform_weights(
+        self,
+        weights: sc.Variable | sc.DataArray | None = None,
+        *,
+        threshold: float = 0.1,
+    ) -> sc.DataArray:
+        """
+        Transform raw pixel weights to the projection plane.
+
+        Parameters
+        ----------
+        weights:
+            Raw pixel weights to transform. If None, default weights of 1 are used.
+        threshold:
+            Threshold for identifying bins with a low weight. If the weight is below the
+            threshold times the median weight, the bin is marked as invalid. This is
+            relevant to avoid issues with color scales in plots, where noise in bins
+            with low weight may dominate the color scale if auto-scaling is used.
+        """
+        if weights is None:
+            weights = sc.ones(
+                sizes=self.detector_number.sizes, dtype='float32', unit=''
+            )
+        else:
+            if weights.sizes != self.detector_number.sizes:
+                raise sc.DimensionError(
+                    f'Invalid {weights.sizes=} for {self.detector_number.sizes=}.'
+                )
+            if (
+                isinstance(weights, sc.DataArray)
+                and (detector_number := weights.coords.get('detector_number'))
+                is not None
+            ):
+                if not sc.identical(detector_number, self.detector_number):
+                    raise sc.CoordError("Mismatching detector numbers in weights.")
+                weights = weights.data
```

Comment:
This is missing a branch for when there is no 'detector_number' coord. In that case, `weights` remains a `DataArray` and it doesn't look like the code below can handle this properly.

Reply:
👍